crates/nvngx-sys/build.rs

```diff
@@ -65,19 +65,25 @@ fn main() {
         // included header files changed.
         .parse_callbacks(Box::new(bindgen::CargoCallbacks::new()))
         // Types and functions defined by the SDK:
-        .allowlist_item("NVSDK_NGX_\\w+")
+        .allowlist_item(r"(PFN_)?NVSDK_NGX_\w+")
         // Single exception for a function that doesn't adhere to the naming standard:
         .allowlist_function("GetNGXResultAsString")
         // Exportable symbols defined by our `bindings.c/h`, wrapping `static inline` helpers
-        .allowlist_function("HELPERS_NGX_\\w+")
+        .allowlist_function(r"HELPERS_NGX_\w+")
+        // Disallow DirectX and CUDA APIs, for which we do not yet provide/implement bindings
+        .blocklist_item(r"\w+D3[Dd]1[12]\w+")
+        .blocklist_type("PFN_NVSDK_NGX_ResourceReleaseCallback")
+        .blocklist_item(r"\w+CUDA\w+")
+        // Disallow all other dependencies, like those from libc or Vulkan.
+        .allowlist_recursively(false)
```

Comment:
Fwiw this makes `bindgen` not generate transitive dependencies, unless they match one of the `allowlist` regexes above. Allowing us to be very precise in the bindings that are generated and where the dependencies originate from.

At this point I'm still not 100% confident that this generated file will be identical between Windows and Linux, but that's what we strive for rather than having copies per target (where it's once again easier to just run `bindgen` based on the target..).